src/standalone/Soil/energy_hydrology.jl

```diff
@@ -108,6 +108,8 @@ struct EnergyHydrology{FT, PS, D, BCRH, S} <: AbstractSoilModel{FT}
     sources::S
     "A boolean flag which, when false, turns off the horizontal flow of water and heat"
     lateral_flow::Bool
+    "Components"
+    land_components::Tuple
```

Comment:
since EnergyHydrology is a standalone model rather than integrated, does it need land_components? I thought it would be used only for integrated models to keep track of the component models

Reply:
I think this is a good question!

we somehow need to indicate to `soil_boundary_fluxes!` how it is being run (in a standalone model, or in an integrated model, with which components). we could stick the `components` into the AtmosDrivenFluxBC type, but that has the same philosophical issue in a way (the bc is part of the model, even in standalone runs you would define the land_components). But maybe that is a little preferable? 

What I had before was different BC types ("AtmosDrivenFluxBC", "AtmosDrivenFluxBCWithSnow", etc). I like this the least because it requires us to come up with names.

Another option is to have the soil_boundary_fluxes! update computation be explicitly moved outside of the `soil_boundary_fluxes! and into this one:
`https://github.com/CliMA/ClimaLand.jl/blob/145b001d3678a13876d3fe537f6ec9d755cab78f/src/integrated/soil_canopy_model.jl#L252` because this is made using the type of the land model, which knows what components are present. Then `soil_boundary_fluxes!` for all "IntegratedLandModelBC" types would do nothing, as p.soil.top_bc would already have been set by the land `update_boundary_fluxes!` function. This is what we were talking about Monday with Gabriele, as I understood it.

Reply:
The way it is right now, the `land_components` default is (:soil,), so we can still run standalone runs in the same way. but youre right that it is a bit odd

Reply:
After discussing with @juliasloan25, we added the field `prognostic_land_components` to AtmosDrivenFluxBC. In a future PR we might rename this to something else, and we would plan to introduce analogs for Snow and Canopy